app/controlplane/internal/service/attestation.go

```diff
@@ -361,6 +361,31 @@ func (s *AttestationService) GetUploadCreds(ctx context.Context, req *cpAPI.Atte
 	return &cpAPI.AttestationServiceGetUploadCredsResponse{Result: resp}, nil
 }
 
+// completeRobotAccount auxiliary method to complete a RobotAccount instance. This is needed since when coming from
+// an API Token request, the only information we have set is the workflow name not the ID
+func (s *AttestationService) completeRobotAccount(ctx context.Context, workflowName string) (*usercontext.RobotAccount, error) {
+	robotAccount := usercontext.CurrentRobotAccount(ctx)
+	if robotAccount == nil {
+		return nil, errors.NotFound("not found", "neither robot account nor API token found")
+	}
+
+	// If WorkflowID is not set and no workflowName is provided, return BadRequest.
+	if workflowName == "" && robotAccount.WorkflowID == "" {
+		return nil, errors.BadRequest("bad request", "when using an API Token, workflow name is required as parameter")
+	}
+
+	// If WorkflowID is not set, retrieve it using the workflow name.
+	if robotAccount.WorkflowID == "" {
+		workflow, err := s.workflowUseCase.FindByName(ctx, workflowName)
```

Comment:
For the case where we want to find the workflow from the name, we need to take into account the organization, since organization is the only element that we can trust (it came from the token) in the request.

otherwise, people could just circumvent these API calls, by providing other orgs names right?

Reply:
I've replaced it by the new `FindByNameInOrg` which does exactly that. Takes the orgID from the token.